Add owner and get items by user

Nw84 opened the pull request.

Nw84 committed:
I have added paths and functions etc so that you can now add a owner to a item and search for items based on the owners/users id

Nw84 committed:
added error handling to getItemsByUserId and addOwner

Nw84 committed:
I have merged my local branch with the main branch

mhema42 commented on src/main/java/com/example/project_spring_boot/controller/ItemController.java:
```diff
@@ -29,9 +31,19 @@ public ResponseEntity <List<Item>> getItems() {
         return new ResponseEntity<>(itemService.createItem(item), HttpStatus.CREATED);
      }
+
+     @PutMapping("/item/{itemId}")
```
I know we talked briefly about put and patch before, shouldn´t it be considered to be a partial change to item? and therefore be patch?
![image](https://user-images.githubusercontent.com/89969770/207024689-439e5432-6c80-420b-92af-6041e8fdff2c.png)

Nw84 replied:
You are right, as usual ;) I will change to patch right away :)

mhema42 reviewed (approved):
Looks very good Tobias, and works fine to.
But I left a comment in ItemController if it should be patch instead of put when it only is a partial update to a item? 
Nice job as allways!

Nw84 committed:
I changed the http method from put to patch after feedback from Mats

Nw84 committed:
i changed the path to getItemsByUserid and addowner and to requestparam instaed of pathvariable, to make it easier for the frontend

Nw84 commented:
> Looks very good Tobias, and works fine to. But I left a comment in ItemController if it should be patch instead of put when it only is a partial update to a item? Nice job as allways!

Fixed! But I also changed the paths to the addowner and getItemByUserId and changed from pathvariable to requestparam to make it easier and more understandable for the frontend. What do you think ?

mhema42 reviewed (approved):
Yes, the new path and using parameter for userId and itemId makes it clearer.
Nice!

emmaambr reviewed (approved):
💯
